### PR TITLE
Adds support for Swift Package Manager

### LINKED ISSUE #23: Support for Swift Package Manager
It would be great if this library can have SPM support.

### PR DESCRIPTION
**Pull request by robcas3**
## Checklist
- [x] All tests are passed.  (https://developer.apple.com/library/mac/documentation/Xcode/Reference/xcode_markup_formatting_ref).  
- [x] Searched [existing pull requests](https://github.com/ra1028/Carbon/pulls) for ensure not duplicated.  

Fixes #23

### CODE REVIEW

File: Carbon.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -655,6 +655,7 @@
 					"@executable_path/Frameworks",
 					"@loader_path/Frameworks",
 				);
+				MARKETING_VERSION = 0.3;
```
Comment (ra1028) [inline Carbon.xcodeproj/project.pbxproj]: Is this a necessary change?

File: Carbon.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -685,6 +686,7 @@
...
```
Comment (ra1028) [inline Carbon.xcodeproj/project.pbxproj]: ditto

File: Package.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+let package = Package(
+    name: "Carbon",
+    platforms: [
+        .iOS(.v10)
+    ],
+    products: [
+        .library(name: "Carbon", targets: ["Carbon"]),
+    ],
+    dependencies: [
+        .package(url: "https://github.com/ra1028/DifferenceKit.git", from: "1.1.2")
```
Comment (ra1028) [inline Package.swift]: .package(url: "https://github.com/ra1028/DifferenceKit.git", .upToNextMinor(from: "1.1.3"))

File: Sources/Info.plist [hunk 4]
```diff
@@ -15,7 +15,7 @@
 	<key>CFBundlePackageType</key>
 	<string>FMWK</string>
 	<key>CFBundleShortVersionString</key>
-	<string>1.0</string>
+	<string>$(MARKETING_VERSION)</string>
```
Comment (ra1028) [inline Sources/Info.plist]: ditto

File: Package.swift [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import PackageDescription
+
+let package = Package(
+    name: "Carbon",
+    platforms: [
+        .iOS(.v10)
+    ],
+    products: [
+        .library(name: "Carbon", targets: ["Carbon"]),
```
[the remark this answers is not in the document]
Comment (ra1028) [reply Package.swift]: Could you fix the commented issues from HoundCI?

File: Carbon.xcodeproj/project.pbxproj (hunk 1 above)
Comment (robcas3) [reply Carbon.xcodeproj/project.pbxproj]: I think this is a Xcode 11 thing. I did not notice it when I committed. Should I remove it?

Comment (ra1028) [reply Carbon.xcodeproj/project.pbxproj]: It should be erased if it works properly without it, IMHO.

**Comment by robcas3** [comment]
Thank you!

**Pull request merged**